test/test_graph_tools.py

```diff
@@ -222,6 +222,24 @@ def test_induced_subgraph():
     assert subgraph == expected_subgraph
 
 
+def test_prioritzed_topological_sort():
+
+    from pytools.graph import compute_topological_order
+
+    priorities = {'a': 1, 'b': 2, 'c': 3, 'e': 4, 'd': 1}
+    dag = {
+            'a': ['b', 'c'],
+            'b': [],
+            'c': ['d', 'e'],
+            'd': [],
+            'e': []}
+
+    def key(u):
+        return priorities[u]
+
+    assert compute_topological_order(dag, key=key) == ['a', 'c', 'e', 'b', 'd']
+
+
```

Comment:
I don't find this test convincing. Sure, it can live. But I would prefer a randomly generated graph with randomly generated priorities and a test that verifies the invariant satisfied by the topsort.

Reply:
Added some more generalized tests.

Reply:
Thanks! Those look good.